source/sundials/sunlinsol_wrapper.cc

```diff
@@ -272,8 +281,13 @@ namespace SUNDIALS
     int
     arkode_linsol_set_preconditioner(SUNLinearSolver LS,
                                      void           *P_data,
-                                     PSetupFn        p_setup,
-                                     PSolveFn        p_solve)
+#  if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+                                     SUNPSetupFn p_setup,
+                                     SUNPSolveFn p_solve)
+#  else
+                                     PSetupFn p_setup,
+                                     PSolveFn p_solve)
+#  endif
```

Comment:
Same here. I have a preference to keep parentheses balanced, which would suggest moving them out.